proposals/NNNN-ios-check-type.md

```diff
@@ -0,0 +1,82 @@
+```
+# Feature name
+
+* Proposal: [SDL-NNNN](NNNN-ios-check-type.md)
+* Author: [Misha Vyrko](https://github.com/mvyrko)
+* Status: **Awaiting review**
+* Impacted Platforms: [iOS]
+
+## Introduction
+
+Method (`-(nullable id)sdl_objectForName:(SDLName)name`) in NSDictionary (Store) returns objects as NSObject. 
+
+Clients(`SDLRegisterAppInterface`, `SDLChangeRegistration` and etc) force cast(without checking) the value(NSObject) from the method to type that they want to have NSNumber, NSString, SDLLanguage and etc.
+I suggest to require to specify the type explicitly.
+Remove method: `- (nullable id)sdl_objectForName:(SDLName)name;`
+Use existing method: `- (nullable id)sdl_objectForName:(SDLName)name ofClass:(Class)classType;` 
+
+
+## Motivation
+
+Returned value type isn't checked at compile time as result we have a lot of crashes in app where one type replaced by another.
+`-[NSNull count]: unrecognized selector sent to instance 0x1e99699d0`
+`-[__NSDictionaryI firstObject]: unrecognized selector sent to instance 0x282447e80`
+`-[__NSCFNumber isEqualToEnum:]: unrecognized selector sent to instance 0xecbbf7f37a5c3eb2`
+
+Checking type at compile time increase stability and resolve problems with unrecognized selector sent to instance.
+
+
+## Proposed solution
+
+Remove method `-(nullable id)sdl_objectForName:(SDLName)name;`. 
```

Comment:
Update to: 

**The proposed solution** is to remove method `-(nullable id)sdl_objectForName:(SDLName)name;`.